openmc/mgxs/library.py

```diff
@@ -717,3 +721,267 @@ def load_from_file(filename='mgxs', directory='mgxs'):
+                if 'kappa-fission' in self.mgxs_types:
+                    mymgxs = self.get_mgxs(domain, 'kappa-fission')
+                    xsdata.set_kappa_fission_mgxs(mymgxs, xs_type=xs_type,
+                                                  nuclide=nuclide)
+                if 'chi' in self.mgxs_types:
+                    mymgxs = self.get_mgxs(domain, 'chi')
+                    xsdata.set_chi_mgxs(mymgxs, xs_type=xs_type,
+                                        nuclide=nuclide)
+                if 'nu-fission' in self.mgxs_types:
+                    mymgxs = self.get_mgxs(domain, 'nu-fission')
+                    xsdata.set_nu_fission_mgxs(mymgxs, xs_type=xs_type,
+                                               nuclide=nuclide)
+                # multiplicity requires scatter and nu-scatter
+                if ((('scatter matrix' in self.mgxs_types) and
+                     ('nu-scatter matrix' in self.mgxs_types))):
+                    scatt_mgxs = self.get_mgxs(domain,
+                                               'scatter matrix')
+                    nuscatt_mgxs = self.get_mgxs(domain,
+                                                 'nu-scatter matrix')
+                    xsdata.set_multiplicity_mgxs(nuscatt_mgxs, scatt_mgxs,
+                                                 xs_type=xs_type,
+                                                 nuclide=nuclide)
+                    using_multiplicity = True
+                else:
+                    using_multiplicity = False
+
+                if using_multiplicity:
+                    nuscatt_mgxs = self.get_mgxs(domain,
+                                                 'nu-scatter matrix')
+                    xsdata.set_scatter_mgxs(nuscatt_mgxs, xs_type=xs_type,
+                                            nuclide=nuclide)
+                else:
+                    if 'nu-scatter matrix' in self.mgxs_types:
+                        nuscatt_mgxs = self.get_mgxs(domain,
+                                                     'nu-scatter matrix')
+                        xsdata.set_scatter_mgxs(nuscatt_mgxs, xs_type=xs_type,
+                                                nuclide=nuclide)
+
+                        # Since we are not using multiplicity, then
+                        # scattering multiplication (nu-scatter) must be
+                        # accounted for approximately by using an adjusted
+                        # absorption cross section.
+                        if 'total' in self.mgxs_types:
+                            xsdata._absorption = \
+                                np.subtract(xsdata.total,
+                                            np.sum(xsdata.scatter[0, :, :],
+                                                   axis=1))
+
+                xsdatas.append(xsdata)
+
+        # Add XSdatas to file
+        mgxs_file.add_xsdatas(xsdatas)
+
+        # Finally, write the file
+        mgxs_file.export_to_xml(full_filename)
```

Comment:
I'm inclined to think that we should leave the XML generation part up to the user and simply return the `mgxs_file`. This would allow us to simplify the method signature by eliminating the `return_names` parameter since the names would be included in the returned `mgxs_file` anyway. What do you think?